server/src/routes/flashcard.ts

```diff
@@ -0,0 +1,56 @@
+import express, { Request, Response } from 'express';
+import { Flashcard, validateFlashcard } from '../models/Flashcards';
+import { FlashcardCollection } from '../models/FlashcardCollection';
+const router = express.Router();
+
+router.put('/:id', async (req: Request, res: Response) => {
+    const {error} = validateFlashcard(req.body);
+    if(error) return res.status(400).send(error.details[0].message);
+
+    const collection = await FlashcardCollection.findById(req.body.params.id);
+    if(!collection) return res.status(404).send('There is no flashcard collection with a given ID');
+
+    let flashcard = new Flashcard({
+        FlashcardCollection: {//tu pytanie, czy nie potrzebujemy referencji w modelu flashcard
+        _id: FlashcardCollection._id,
+        name: FlashcardCollection.name
+        },
+        prompt: req.body.prompt,
+        imageUrl: req.body.imageUrl,
+        answer: req.body.answer,
+        extraInfo: req.body.extraInfo
+    }) 
+
+    flashcard = await flashcard.save();
```

Comment:
Nie bardzo rozumiem po co robimy ponowne przypisanie po zrobieniu save, nie wystarczy po prostu?:  
```suggestion
    await flashcard.save();
```